azkaban-common/src/main/java/azkaban/utils/FileIOUtils.java

```diff
@@ -42,6 +42,31 @@
 public class FileIOUtils {
   private final static Logger logger = Logger.getLogger(FileIOUtils.class);
 
+  /**
+   * Check if a directory is writable
+   *
+   * @param dir directory file object
+   * @return true if it is writable. false, otherwise
+   */
+  public static boolean isDirWritable(File dir) {
```

Comment:
spoke offline. This can be unit tested by creating a dir and altering permissions using OS calls and verifying that we get the required results.